src/go/rpk/pkg/cli/cmd/acl/common.go

```diff
@@ -57,26 +58,90 @@ type (
 	// Corresponding to the above, acl and aclWithMessage are the rows
 	// for PrintStructFields.
 	acl struct {
-		Principal           string
-		Host                string
-		ResourceType        kmsg.ACLResourceType
-		ResourceName        string
-		ResourcePatternType kmsg.ACLResourcePatternType
-		Operation           kmsg.ACLOperation
-		Permission          kmsg.ACLPermissionType
+		Principal           string                      `json:"principal" yaml:"principal"`
+		Host                string                      `json:"host" yaml:"host"`
+		ResourceType        kmsg.ACLResourceType        `json:"resource_type" yaml:"resource_type"`
+		ResourceName        string                      `json:"resource_name" yaml:"resource_name"`
+		ResourcePatternType kmsg.ACLResourcePatternType `json:"resource_pattern_type" yaml:"resource_pattern_type"`
+		Operation           kmsg.ACLOperation           `json:"operation" yaml:"operation"`
+		Permission          kmsg.ACLPermissionType      `json:"permission" yaml:"permission"`
 	}
+
 	aclWithMessage struct {
-		Principal           string
-		Host                string
-		ResourceType        kmsg.ACLResourceType
-		ResourceName        string
-		ResourcePatternType kmsg.ACLResourcePatternType
-		Operation           kmsg.ACLOperation
-		Permission          kmsg.ACLPermissionType
-		Message             string
+		Principal           string                      `json:"principal" yaml:"principal"`
+		Host                string                      `json:"host" yaml:"host"`
+		ResourceType        kmsg.ACLResourceType        `json:"resource_type" yaml:"resource_type"`
+		ResourceName        string                      `json:"resource_name" yaml:"resource_name"`
+		ResourcePatternType kmsg.ACLResourcePatternType `json:"resource_pattern_type" yaml:"resource_pattern_type"`
+		Operation           kmsg.ACLOperation           `json:"operation" yaml:"operation"`
+		Permission          kmsg.ACLPermissionType      `json:"permission" yaml:"permission"`
+		Message             string                      `json:"message" yaml:"message"`
```

Comment:
For marshaling, this can be `json:"error,omitempty" yaml:"error,omitempty"`

I forget why I chose the message column originally, but we can always change the text output to be consistent later if need be (not in this PR)